src/main/java/seedu/duke/tasks/ModuleList.java

```diff
@@ -21,6 +21,14 @@ public Module addModule(Module m) {
         return m;
     }
 
+    /**
+     * Removes specified module from the module list.
+     * @param moduleCode the module code to be removed
+     */
+    public void removeModule(String moduleCode) {
+        list.remove(getModule(moduleCode));
```

Comment:
Maybe check whether `getModule()` returns `null` before attempting to call `list.remove()` on it.

Also, see above remark on returning the removed module instead.